src/polybench/POLYBENCH_2MM-Sycl.cpp

```diff
@@ -12,116 +12,96 @@
 
 #if defined(RAJA_ENABLE_SYCL)
 
+#include "common/SyclDataUtils.hpp"
+
 #include <iostream>
 #include <cmath>
 
-#include "common/SyclDataUtils.hpp"
-
 namespace rajaperf 
 {
 namespace polybench
 {
 
   //
-  // Define thread block size for SYCL execution
+  // Define work-group shape for SYCL execution
   //
-  const size_t block_size = 16;
-
-#define POLYBENCH_2MM_DATA_SETUP_SYCL \
-  allocAndInitSyclDeviceData(tmp, m_tmp, m_ni * m_nj, qu); \
-  allocAndInitSyclDeviceData(A, m_A, m_ni * m_nk, qu); \
-  allocAndInitSyclDeviceData(B, m_B, m_nk * m_nj, qu); \
-  allocAndInitSyclDeviceData(C, m_C, m_nj * m_nl, qu); \
-  allocAndInitSyclDeviceData(D, m_D, m_ni * m_nl, qu); \
-\
-  Real_type alpha = m_alpha; \
-  Real_type beta = m_beta; \
-
-
-#define POLYBENCH_2MM_TEARDOWN_SYCL \
-  getSyclDeviceData(m_D, D, m_ni * m_nl, qu); \
-  deallocSyclDeviceData(tmp, qu); \
-  deallocSyclDeviceData(A, qu); \
-  deallocSyclDeviceData(B, qu); \
-  deallocSyclDeviceData(C, qu); \
-  deallocSyclDeviceData(D, qu);
-
-void POLYBENCH_2MM::runSyclVariant(VariantID vid)
+#define in_wg_sz (32)
+#define out_wg_sz (work_group_size / in_wg_sz)
+
+
+template <size_t work_group_size >
+void POLYBENCH_2MM::runSyclVariantImpl(VariantID vid)
 {
-  const unsigned long run_reps = getRunReps();
+  const Index_type run_reps = getRunReps();
 
   auto res{getSyclResource()};
   auto qu = res.get_queue();
 
   POLYBENCH_2MM_DATA_SETUP;
 
   if ( vid == Base_SYCL ) {
-    {
-      POLYBENCH_2MM_DATA_SETUP_SYCL;
-
-      const size_t ni_grid_size = block_size * RAJA_DIVIDE_CEILING_INT(m_ni, block_size);
-      const size_t nj_grid_size = block_size * RAJA_DIVIDE_CEILING_INT(m_nj, block_size);
-      const size_t nl_grid_size = block_size * RAJA_DIVIDE_CEILING_INT(m_nl, block_size);
-
-      startTimer();
-      for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
-
-        qu->submit([&] (sycl::handler& h) {
-          h.parallel_for<class polybench2MM_1>(sycl::nd_range<2> 
-                                               {sycl::range<2> {ni_grid_size, nj_grid_size},
-                                                sycl::range<2> {block_size, block_size}},
-                                               [=] (sycl::nd_item<2> item) {
-
-           Index_type i = item.get_global_id(0); 
-           Index_type j = item.get_global_id(1); 
-
-           if (i < ni && j < nj) {
-             POLYBENCH_2MM_BODY1;
-             for (Index_type k=0; k < nk; ++k) {
-                POLYBENCH_2MM_BODY2;
-              }
-              POLYBENCH_2MM_BODY3;
+
+    sycl::range<3> global_dim1(1,
```

Comment:
Don't we only need a 2d range?

Reply:
the RAJA backend uses a 3D range

Reply:
Technically, yes, only a 2d range is needed. I discussed the pattern and consistency across kernels with @homerdin and we agreed that this is how we should do it.

Reply:
There are places where we use kernel for a 1d loop and then the base variant uses a 1d range instead of a 3d range, it makes sense to me to use what you would naturally use instead of doing what RAJA::kernel happens to use. If it ends up making a difference we could see it.